habitat/tasks/rearrange/multi_task/composite_sensors.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from habitat.core.embodied_task import Measure
+from habitat.core.registry import registry
+
+
+@registry.register_measure
+class CompositeReward(Measure):
+    cls_uuid: str = "composite_reward"
+
+    @staticmethod
+    def _get_uuid(*args, **kwargs):
+        return CompositeReward.cls_uuid
+
+    def __init__(self, sim, config, *args, **kwargs):
+        super().__init__(**kwargs)
+        self._sim = sim
+        self._config = config
+        self._prev_node_idx = None
+
+    def reset_metric(self, *args, episode, task, observations, **kwargs):
+        task.measurements.check_measure_dependencies(
+            self.uuid,
+            [CompositeNodeIdx.cls_uuid, CompositeSuccess.cls_uuid],
+        )
+
+        self.update_metric(
+            *args,
+            episode=episode,
+            task=task,
+            observations=observations,
+            **kwargs,
+        )
+
+    def update_metric(self, *args, episode, task, observations, **kwargs):
+        reward = 0.0
+        node_measure = task.measurements.measures[CompositeNodeIdx.cls_uuid]
+
+        node_idx = node_measure.get_metric()["node_idx"]
+        if self._prev_node_idx is None:
+            self._prev_node_idx = node_idx
+        elif node_idx > self._prev_node_idx:
+            reward += self._config.STAGE_COMPLETE_REWARD
+
+        cur_task = task.get_cur_task()
+        if cur_task is None:
+            cur_task_cfg = task.get_inf_cur_task()._config
+        else:
+            cur_task_cfg = cur_task._config
+
+        if "REWARD_MEASUREMENT" not in cur_task_cfg:
+            raise ValueError("Cannot find REWARD_MEASUREMENT key")
```

Comment:
Wouldn't the following line also raise a similar error if the key was not present? What is the reason behind checking for this particular key and not the others?

Reply:
`REWARD_MEASUREMENT` is not defined in a default.py

The `cls_uuid`'s for dependencies are already checked as dependencies and the `SUCCESS_REWARD` is defined in a default.py